src/hostcalls.rs

```diff
@@ -25,7 +26,10 @@ pub fn log(level: LogLevel, message: &str) -> Result<(), Status> {
     unsafe {
         match proxy_log(level, message.as_ptr(), message.len()) {
             Status::Ok => Ok(()),
-            status => panic!("unexpected status: {}", status as u32),
+            status => {
+                error!("[log] unexpected status: {}", status as u32);
+                Err(status)
+            },
```

Comment:
This is done by design, since those errors are illegal are not actionable. See #282.